### ISSUE TITLE
Multiple dimensions support for `torch.max`

### ISSUE THREAD

**Issue by lukas-folle-snkeos** [pytorch/pytorch#105053]
### 🚀 The feature, motivation and pitch

Numpy's  `np.max` has support for multiple dimensions, but `torch.max` does not.
To keep both synced and avoid implementing custom function like #105045, it would be really helpful to have this.

### Alternatives

#105045

**Comment by lukas-folle-snkeos** [comment]
Thanks for getting back to me @albanD :blush:
> You can use torch.amax() that does support multiple dims though:

That's true. The only thing I am missing with `torch.amax` is retrieving the indices of the maximum locations. My use case is described in #105045. 

Currently, I have do to something like `torch.nonzero(torch.Tensor(input == torch.amax(input, dims, keepdim=True)))` which seems unnecessarily complex to me.

**Comment by albanD** [comment]
Ho I didn't check the other issue, I though it was an old one where we added multidim support to argmax haha
I'm not sure I follow your example there. In particular, what should be the output shape? I guess `(3, 1, 62)` ? Where `(1, 62)` are the preserved dims and `3` because we collapsed 3 dimensions?

**Comment by lukas-folle-snkeos** [comment]
Yes, something like that!

If I'd do
```python
input= torch.rand((1, 66, 64, 64, 64))
values = torch.nonzero(torch.Tensor(input == torch.amax(input, (2, 3, 4), keepdim=True))
```
I would expect this result: `values.shape (66, 5)`
Which holds the location of the maxima in each of the 66 channels.

**Comment by albanD** [comment]
But why 66 and not `(1, 66)`? Also why 5 and not 3?

**Comment by lukas-folle-snkeos** [comment]
You are right, `(1, 66)` makes more sense. I thought 5 since those are the indices that can be applied as is to the tensor. But 3 would also make sense to me, since this is the number of requested dims.

**Comment by lukas-folle-snkeos** [comment]
@albanD what do think of this?

**Comment by albanD** [comment]
Hey!
I'm not sure tbh. The fact that there are multiple possible options is a problem as we need to take the time to look at them in details to see what to do. Hence the "needs design" label. But I'm not sure anyone on our end is looking into this.

**Comment by lukas-folle-snkeos** [comment]
@albanD If there was an unravel index in torch this would also work with the plain argmax implementation of torch.

**Comment by albanD** [comment]
Ho the unravel impl has an interesting history, let's see if we can revive this one as it will be less ambiguous (we just want to match numpy) https://github.com/pytorch/pytorch/issues/35674

**Comment by lukas-folle-snkeos** [comment]
@albanD just following up on this. 
With the availability of the unravel_index function we can close this, right?
